substrate/primitives/state-machine/src/lib.rs

```diff
@@ -289,7 +289,7 @@ mod execution {
 
 			let result = self
 				.exec
-				.call(&mut ext, self.runtime_code, self.method, self.call_data, false, self.context)
```

Comment:
CI tests are failing because of this change. 
Previously (on master) we were always executing wasm runtime here, in particular this sequence was using wasm runtime: `sc_consensus_grandpa::aux_schema::load_persistent` -> `sc_consensus_grandpa::GenesisAuthoritySetProvider<Block>>::get`.

With this change the native runtime is executed. Since this change https://github.com/paritytech/polkadot-sdk/commit/ebcf0a0f1cab2d43718ba96d26e5687f4d14580a is not compatible with old state layouts, the tests are failing.

I opened https://github.com/paritytech/polkadot-sdk/pull/2521 which gets rid of native runtime in substrate-node. Once merged we can proceed with this one.